### DISCUSSION TITLE
Custom operator affects prevoius operator.

### DISCUSSION THREAD

**Discussion by Totati** [ReactiveX/rxjs#6484] [Q&A] [answered]
Hi, I'm trying to make custom operators, but it messes up typings. What am I missing or doing wrong?
My operator looks like this:
``` typescript
export function isBlack(): OperatorFunction<Fruit | Fish, boolean> {
  return source => source.pipe(map(v => v.color === 'black'));
}
```
When i use it in a pipe, it messes up the previous tap's typing. 
[Stackblitz example](https://stackblitz.com/edit/rxjs-sutm5w)

**Comment by raymonddavis** [comment, marked as answer]
The typing is incorrect on your custom operators.

```ts
export function isBlack(): OperatorFunction<Fruit | Fish, boolean> {
  return source => source.pipe(map(v => v.color === 'black'));
}
```
This implies that your operator will return `Fruit | Fish`.
What i think you want to do is this
```ts
export function isBlack<T extends Fruit | Fish>(): OperatorFunction<T, boolean> {
  return source => source.pipe(map(v => v.color === 'black'));
}
```
This allows `Fruit | Fish` to be passed in and whatever one is passed is also returned.

Your previous example will allow anything to be passed in and `hard` setting the return to a "new" type `Fruit | Fish`

Updated typing
[Stackblitz example](https://stackblitz.com/edit/rxjs-b4m2nl?file=index.ts)

**Reply by Totati** [reply]
Thank you, yes I wanted this :)
